### ISSUE TITLE
Allowed to add invalid credentials when creating Global DNS Provider.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#17517]
Rancher server version - Build from master 01/22

Steps to reproduce the problem:
Create a Global DNS Provider with invalid Access key /Secret key.
Global DNS Provider get created successfully.

Expected Behavior:
Global DNS Provider creation should fail with error message indicating that credentials are bad.

**Comment by prachidamle** [comment]
I don't think we will be able to get an error message if the keys are invalid. There is no way to validate the user's keys prior to launching the kubernetes/external-dns deployment.

**Comment by alena1108** [comment]
Given that external-dns container won't exit on invalid key, there is nothing Rancher can do at the moment to bubble up the failure to the user. The error could be seen only by looking at the external-dns container logs. Moving out the issue from 2.2
